### PR TITLE
dev/core#2656 Add test to demonstrate issue / bug

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Test demonstrates issue described in https://lab.civicrm.org/dev/core/-/issues/2656

Before
----------------------------------------
No test

After
----------------------------------------
Look look - it's a test and it fails

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Test Result (2 failures / +2)
CRM_Member_BAO_MembershipTest.testMembershipUpdateDoesNotDeleteRelatedMembershipsByMistake
api_v3_ContactTest.testCreateApiKey with data set "APIv4"

**Comment by eileenmcnaughton** [comment]
Both those tests seem like they should have been failing anyway - I've added fixes for them - lets see

File: tests/phpunit/CRM/Member/BAO/MembershipTest.php [hunk 1]
```diff
@@ -40,7 +40,6 @@ public function setUp(): void {
    * @throws \CiviCRM_API3_Exception
    */
   public function tearDown(): void {
-    $this->membershipStatusDelete($this->_membershipStatusID);
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Member/BAO/MembershipTest.php:43]: this cleanup is eventually unrelated - but might as well stay in

**Comment by eileenmcnaughton** [comment]
@colemanw the test for your fix is passing now

**Comment by seamuslee001** [comment]
main fix has been merged and this is test to lock in the fix LGIM

**Pull request merged**